modules/afmongodb/CMakeLists.txt

```diff
@@ -1,3 +1,7 @@
+if (ENABLE_MONGODB STREQUAL DISABLE)
+  return()
+endif()
+
```

Comment:
I think `module_switch` supports disabling modules with the standard `OFF` syntax.

Based on the geoip2 module:
```
find_package(libmongoc-1.0)
module_switch(ENABLE_MONGODB "Enable mongodb destination driver" libmongoc-1.0_FOUND)

if (NOT ENABLE_MONGODB)
  return ()
endif ()

if (NOT libmongoc-1.0_FOUND)
  message(FATAL_ERROR "The mongodb module is enabled but libmongoc not found.")
endif ()

set(AFMONGODB_HEADERS
...
```

```
$ cmake ../ -DIVYKIS_SOURCE=system -DJSONC_SOURCE=system -DSYSLOG_NG_ENABLE_DEBUG=ON -DENABLE_JAVA=OFF -DENABLE_MONGODB=OFF
```

Reply:
Oh I see, CMake fails at `/usr/lib/x86_64-linux-gnu/cmake/libbson-1.0/libbson-1.0-config.cmake:28`. It's really unfortunate, because `find_package()` without the REQUIRED options should not really stop the execution.

Reply:
@MrAnno : exactly... and we need to return _before_ findpackage